go/cmd/dolt/commands/filter-branch.go

```diff
@@ -118,44 +120,73 @@ func (cmd FilterBranchCmd) Exec(ctx context.Context, commandStr string, args []s
 		queryString = string(queryStringBytes)
 	}
 
-	replay := func(ctx context.Context, commit, _, _ *doltdb.Commit) (doltdb.RootValue, error) {
-		var cmHash, before hash.Hash
-		var root doltdb.RootValue
+	replayRootVal := func(ctx context.Context, root, _, _ doltdb.RootValue) (doltdb.RootValue, error) {
```

Comment:
Rather than having multiple anonymous functions here, define interfaces like:

```go
type CommitReplayer interface {
     Replay(ctx context.Context, root, _, _ doltdb.RootValue) (doltdb.RootValue, error)
}
```

Then define a type in this file that has the replay logic as functions, and pass that type through to the Rebase functions

Reply:
I believe the anonymous functions were used to avoid import cycles. `rebase/filter_branch.go` would have to import `command` package if I create the interface here.

Reply:
Create the interface in the rebase package and use it in the command package.